src/core_atmosphere/physics/physics_wrf/module_cu_gf_deep.F

```diff
@@ -2304,23 +2306,44 @@ subroutine cu_gf_deep_run(        &
           endif
       enddo
       !-----downdraft_cold_pool---------------
-      if (icoldpool > 0 .and. imid ==0) then
+      !if (icoldpool > 0 .and. imid ==0) then
+      if (icoldpool /= 0 .and. icoldpool /= 2 .and. imid ==0) then
       ! --- adding the gust front horizontal speed to the 2-d MCL wind
       ! --- only magnitude is augmented, direction is kept the same
       do i=its,itf
+          umcl(i)=0.
+          vmcl(i)=0.
+          dpsum=0.
           if(ierr(i) > 0 ) cycle
+           do k=kts+1,ktop(i)-1
+             trash =-(po_cup(i,k)-po_cup(i,kts))
+             if(trash.gt.300. .and. trash.lt.600.)then
+             dp=100.*(po_cup(i,k)-po_cup(i,k+1))
+             umcl(i)=umcl(i)+us(i,k)*dp
+             vmcl(i)=vmcl(i)+us(i,k)*dp
+             dpsum=dpsum+dp
+             endif
+           enddo
+           if(dpsum > 0.) then
+           umcl(i)=umcl(i)/dpsum
+           vmcl(i)=vmcl(i)/dpsum
            MCL_speed= sqrt( umcl(i)**2 + vmcl(i)**2 )
            aux =  (MCL_speed + vcpool(i))/(MCL_speed+1.e-6)
            umcl(i) = aux * umcl(i)
            vmcl(i) = aux * vmcl(i)
+           endif
       enddo
       ! --- gust front momentum impact
       do i=its,itf
-          if(ierr(i) > 0 ) cycle
-           do k=kts,kdet(i)
+          if(ierr(i) > 0 .or. vcpool(i) .le.0.) cycle
+             k=kts
              dp=100.*(po_cup(i,k)-po_cup(i,k+1))
-             outu(i,k) = outu(i,k) + edto(i)*zdo(i,k+1)*(umcl(i)-us(i,k))*g/dp*xmb(i)
-             outv(i,k) = outv(i,k) + edto(i)*zdo(i,k+1)*(vmcl(i)-vs(i,k))*g/dp*xmb(i)
+             outu(i,k) = outu(i,k) + edto(i)*zdo(i,k+1)*umcl(i)*g/dp*xmb(i)
+             outv(i,k) = outv(i,k) + edto(i)*zdo(i,k+1)*vmcl(i)*g/dp*xmb(i)
+                do k=kts+1,kdet(i)
```

Comment:
indenting is messed up here

Reply:
Updated following your suggestions. Thanks.